Makefile

```diff
@@ -21,12 +21,14 @@ clean: ## remove all the unneeded artifacts
 	-rm -rf .coverage .coverage.* htmlcov
 
 requirements: ## install the developer requirements
+	pip install -e .
 	pip install -r requirements/dev.txt
 
 upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -qr requirements/pip-tools.txt
 	# Make sure to compile files after any other files they include!
 	pip-compile -v --upgrade -o requirements/pip-tools.txt requirements/pip-tools.in
+	pip-compile -v --upgrade -o requirements/base.txt requirements/base.in
```

Comment:
@nedbat: Any idea why `base.in` wasn't added when this first started using pip-compile?

Reply:
Here is the original conversation: https://github.com/edx/edx-lint/pull/103#discussion_r329625807

Reply:
Still not quite following what has changed between now and then that this couldn’t have been the original solution?

Also, Cale had a comment around pinning pylint that probably should move to a comment next to the pin.

Reply:
Gah. Hoisted by my own petard here, it looks like. I totally forgot the original considerations around this. That said, we should definitely do something so that we can use a standard `make upgrade` flow.